fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -1548,17 +1548,19 @@ ThreadFuture<Void> MultiVersionTransaction::onError(Error const& e) {
 		auto f = tr.transaction ? tr.transaction->onError(e) : makeTimeout<Void>();
 		f = abortableFuture(f, tr.onChange);
 
-		return flatMapThreadFuture<Void, Void>(f, [this, e](ErrorOr<Void> ready) {
-			if (!ready.isError() || ready.getError().code() != error_code_cluster_version_changed) {
-				if (ready.isError()) {
-					return ErrorOr<ThreadFuture<Void>>(ready.getError());
-				}
-
+		return flatMapThreadFuture<Void, Void>(f, [this](ErrorOr<Void> ready) {
+			if (ready.isError() && ready.getError().code() == error_code_cluster_version_changed) {
+				// In case of a cluster version change, upgrade (or downgrade) the transaction
+				// and let it to be retried independently of the original error
+				updateTransaction();
+				return ErrorOr<ThreadFuture<Void>>(Void());
+			}
+			// In all other cases forward the result of the inner onError call
```

Comment:
Do we now never `return ErrorOr<ThreadFuture<Void>>(onError(e));` ?

Reply:
We don't. As you see above we call tr.transaction->onError(e) and forward its result if there is no version change. And in case of a version change we return Void, i.e. success.